src/main/java/com/prgrms/nabmart/domain/item/service/ItemService.java

```diff
@@ -74,6 +75,39 @@ public FindItemsResponse findNewItems(FindNewItemsCommand findNewItemsCommand) {
         return FindItemsResponse.from(items);
     }
 
+    @Transactional
+    public void updateItem(UpdateItemCommand updateItemCommand) {
+        Long itemId = updateItemCommand.itemId();
+        Long mainCategoryId = updateItemCommand.mainCategoryId();
+        Long subCategoryId = updateItemCommand.subCategoryId();
+        
+        Item item = itemRepository.findById(itemId)
+            .orElseThrow(() -> new NotFoundItemException("해당 Id의 아이템은 존재하지 않습니다."));
+        MainCategory mainCategory = getMainCategoryById(mainCategoryId);
+        SubCategory subCategory = getSubCategoryById(subCategoryId);
+        item.updateItem(updateItemCommand.name(), updateItemCommand.price(),
+            updateItemCommand.quantity(), updateItemCommand.description(),
+            mainCategory, subCategory, updateItemCommand.discount());
+    }
+
+    private SubCategory getSubCategoryById(Long subCategoryId) {
+        SubCategory subCategory = null;
+        if (subCategoryId != null) {
+            subCategory = subCategoryRepository.findById(subCategoryId)
+                .orElseThrow(() -> new NotFoundCategoryException("없는 소카테고리입니다."));
+        }
+        return subCategory;
```

Comment:
```suggestion
        return subCategoryRepository.findById(subCategoryId)
                .orElseThrow(() -> new NotFoundCategoryException("없는 소카테고리입니다."));
```
이렇게 수정하시면 Null 체크도 하고 코드가 간결해질 것 같아요!

Reply:
> 이렇게 수정하시면 Null 체크도 하고 코드가 간결해질 것 같아요!

findById에 null이 인자로 전달되면 내부적으로 IllegalArgumentException이 던져서 기존 방식대로 가야할 것 같아요.

Reply:
아하 처음 알았네요! 감사합니다 👍🏻